### PR TITLE
ztp: SiteConfig template overrides

### PR DESCRIPTION
**Pull request by lack**
This adds a new entry to SiteConfig which allows users to override the templates used to construct the generated CRs.

Signed-off-by: Jim Ramsay <jramsay@redhat.com>

### CODE REVIEW

File: ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go [hunk 1]
```diff
@@ -148,9 +179,30 @@ func (scbuilder *SiteConfigBuilder) getClusterCRs(clusterId int, siteConfigTemp
 		}
 	}
 
+	// Double-check that the user didn't ask us to override any invalid object types:
+	err := siteConfigTemp.areAllOverridesValid(&validKinds, &validNodeKinds)
+	if err != nil {
+		return clusterCRs, err
```
[the remark this answers is not in the document]
Comment (lack) [reply ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go:185]: Well, the caller should know that if we return err != nil the rest of the return value is not to be trusted.  So if we return `clusterCRs` or `{}` or `nil` shouldn't make much of a difference...

File: ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go [hunk 2]
```diff
@@ -96,62 +96,142 @@ func (scbuilder *SiteConfigBuilder) validateSiteConfig(siteConfigTemp SiteConfig
...
+	}
+
 	return clusterCRs, nil
 }
 
-func (scbuilder *SiteConfigBuilder) getClusterCR(clusterId int, siteConfigTemp SiteConfig, mapSourceCR map[string]interface{}, nodeId int) (interface{}, error) {
+func (scbuilder *SiteConfigBuilder) instantiateCR(target string, originalTemplate map[string]interface{}, overrideSearch func(kind string) (string, bool), applyTemplate func(map[string]interface{}) (map[string]interface{}, error)) (map[string]interface{}, error) {
+	// Instantiate the CR based on the original original CR
```
Comment (imiller0) [inline ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go:184]: Minor nit: double-original :)

**Pull request merged**